### PR TITLE
Tests for btree implementation used by range trees

### PR DESCRIPTION
**Pull request by jwk404**
Signed-off-by: John Kennedy <john.kennedy@delphix.com>

### Motivation and Context

These are tests for the recently added btree implementation.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by jwk404** [comment]
Looking at the build failures and core dump.

File: tests/zfs-tests/cmd/btree_test/Makefile.am [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2019 by Delphix. All rights reserved.
+#
+
+include $(top_srcdir)/config/Rules.am
```
Comment (behlendorf) [inline tests/zfs-tests/cmd/btree_test/Makefile.am]: I'd suggest unconditionally enabling the ASSERTs for this test case, as was done for `raidz_test`.c.

```
# Unconditionally enable ASSERTs
AM_CPPFLAGS += -DDEBUG -UNDEBUG
```

File: tests/zfs-tests/cmd/btree_test/btree_test.c [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+
+	void *avl_cookie = NULL;
+	while ((node = avl_destroy_nodes(&avl, &avl_cookie)) != NULL)
+		free(node);
+	avl_destroy(&avl);
+
+	return (0);
+}
+
+static int
```
Comment (behlendorf) [inline tests/zfs-tests/cmd/btree_test/btree_test.c]: Can you add a comment explaining this test case like the others.

**Comment by jwk404** [comment]
- use the TREE_CMP macro
- unconditionally enable ASSERTS
- added comments
- changes some types for compilation errors

**Pull request merged**